src/ConnectedMode/ServerSentEvents/QualityProfile/QualityProfileServerEventsListener.cs

```diff
@@ -0,0 +1,57 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.ComponentModel.Composition;
+using System.Threading.Tasks;
+using SonarLint.VisualStudio.ConnectedMode.QualityProfiles;
+
+namespace SonarLint.VisualStudio.ConnectedMode.ServerSentEvents.QualityProfile
+{
+    /// <summary>
+    /// Event listener for <see cref="IQualityProfileServerEventSource"/>
+    /// </summary>
+    internal interface IQualityProfileServerEventsListener
+    {
+        Task ListenAsync();
+    }
+
+    [Export(typeof(IQualityProfileServerEventsListener))]
+    [PartCreationPolicy(CreationPolicy.Shared)]
+    internal class QualityProfileServerEventsListener : IQualityProfileServerEventsListener
+    {
+        private readonly IQualityProfileServerEventSource eventSource;
+        private readonly IQualityProfileUpdater updater;
+
+        [ImportingConstructor]
+        public QualityProfileServerEventsListener(IQualityProfileServerEventSource eventSource, IQualityProfileUpdater updater)
+        {
+            this.eventSource = eventSource;
+            this.updater = updater;
+        }
+
+        public async Task ListenAsync()
+        {
+            while (await eventSource.GetNextEventOrNullAsync() != null)
```

Comment:
Probably worth commenting that the eventSource will return null when it is disposed (same as in the `IssueServerEventListener`).

Reply:
added the comment